lib/src/common/disposable_manager.dart

```diff
@@ -112,6 +134,25 @@ abstract class DisposableManagerV3 implements DisposableManagerV2 {
   Completer<T> manageCompleter<T>(Completer<T> completer);
 }
 
+/// Managers for disposable members.
+///
+/// This interface allows consumers to exercise more control over how
+/// disposal is implemented for their classes.
+///
+/// When new management methods are to be added, they should be added
+/// here first, then implemented in [Disposable].
+abstract class DisposableManagerV4 implements DisposableManagerV3 {
+  /// Automatically cancel a stream subscription when this object is disposed.
+  ///
+  /// If the returned `StreamSubscription` is cancelled manually (i.e. canceled
+  /// before disposal of the parent object) [Disposable] will clean up the
+  /// internal reference allowing the subscription to be garbage collected.
+  ///
+  /// Neither parameter may be `null`.
+  StreamSubscription<T> getManagedStreamSubscription<T>(
+      Stream<T> stream, void onData(T event));
```

Comment:
🔥 You need optional params for `onError`, `onDone`, etc... 

I wonder if the Dart SDK has a typedef for the listen method on streams.